sjsonnet/src/sjsonnet/Val.scala

```diff
@@ -155,10 +155,11 @@ object Val{
 
     private[this] def getValue0: util.LinkedHashMap[String, Obj.Member] = {
       if(value0 == null) {
-        value0 = new java.util.LinkedHashMap[String, Val.Obj.Member]
```

Comment:
Could you add a comment explaining why we do things in this way, and the multi-threaded guarantee we want to maintain?